Formula/unisonlang.rb

```diff
@@ -0,0 +1,78 @@
+require "language/node"
+
+class Unisonlang < Formula
+  desc "Friendly programming language from the future"
+  homepage "https://unison-lang.org/"
+  url "https://github.com/unisonweb/unison.git",
+      tag:      "release/M3",
+      revision: "97cce8d23147df30b0f80ae745968db09f4e1a44"
+  version "M3"
+  license "MIT"
+  head "https://github.com/unisonweb/unison.git", branch: "trunk"
+
+  depends_on "ghc" => :build
+  depends_on "haskell-stack" => :build
+  depends_on "node" => :build
+
+  # stack/ghc currently have a number of issues building for aarch64
+  # https://github.com/unisonweb/unison/issues/3136
+  depends_on arch: :x86_64
+  depends_on "gmp"
+
+  uses_from_macos "xz" => :build
+  uses_from_macos "zlib"
+
+  resource "codebase-ui" do
```

Comment:
I think `gmp` is needed for something `stack` is downloading during build time. It might even be `ghc` itself, but that means it's ignoring the `--no-install-ghc` flag...

```suggestion
  depends_on arch: :x86_64

  uses_from_macos "xz" => :build
  uses_from_macos "zlib"

  on_linux do
    depends_on "gmp" => :build
  end

  resource "codebase-ui" do
```

Reply:
It would be good to understand what exactly is using `gmp`, though.

Reply:
Turns out it was only needed before I realized I was supposed to set those flags for stack 😅 removed now